src/databricks/labs/ucx/workspace_access/clusters.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import InvalidParameterValue
+from databricks.sdk.service.compute import ClusterDetails, DataSecurityMode
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterAccess:
+    def __init__(self, installation: Installation, ws: WorkspaceClient, prompts: Prompts):
+        self._ws = ws
+        self._prompts = prompts
+        self._installation = installation
+
+    def list_cluster(self):
+        cluster_list = {}
+        clusters = self._ws.clusters.list()
+        for cluster in clusters:
+            if cluster.cluster_source is not None and cluster.cluster_source.name != "JOB":
+                cluster_list[cluster.cluster_name] = cluster.cluster_id
```

Comment:
you know that clusters don't have unique names? this means that you can't have them as dict keys

Reply:
Changed the logic. Now hitting the API only once . Using the result to filter out the cluster based on selection